docs(AzureRedis): user docs (#739)
kyma-project/cloud-manager: docs/user/resources/04-80-azure-redis-instance.md

# AzureRedisInstance Custom Resource
The `AzureRedisInstance.cloud-resources.kyma-project.io` is a namespace-scoped custom resource (CR).
It describes the Azure Cache for Redis instance.
Once the instance is provisioned, a Kubernetes Secret with endpoint and credential details is provided in the same namespace.

> By default, the created auth Secret has the same name as the AzureRedisInstance.

Reviewer: ```suggestion
By default, the created auth Secret has the same name as AzureRedisInstance.
```

Author: Fixed.

> The current implementation supports *Premium* tier, which is explained in detail on the [Azure Cache for Redis overview page](https://azure.microsoft.com/en-us/products/cache).

Reviewer: ```suggestion
The current implementation supports the Premium tier, which is explained in detail on the [Azure Cache for Redis overview page](https://azure.microsoft.com/en-us/products/cache).
```

Author: Fixed.

> [!TIP] _Only for advanced cases of network topology_
> 

> > Redis requires 2 IP addresses per shard. *Premium* tear supports [up to 3 shards.](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-high-availability)

Reviewer: ```suggestion
> Redis requires 2 IP addresses per shard. The Premium tier supports [up to 3 shards.](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-high-availability)
```

Author: Fixed.

> IP addresses can be configured via IpRange CR.

Reviewer: ```suggestion
IP addresses can be configured using the IpRange CR.
```

Author: Fixed.

To learn more, read [Configure a reserved IP address range](https://cloud.google.com/filestore/docs/creating-instances#configure_a_reserved_ip_address_range).
Those IP addresses are allocated from the [IpRange CR](./04-10-iprange.md).
If an IpRange CR is not specified in the AzureRedisInstance, then the default IpRange is used.
If the default IpRange does not exist, it is automatically created.
Manually create a non-default IpRange with specified Classless Inter-Domain Routing (CIDR) and use it only in advanced cases of network topology when you want to control the network segments to avoid range conflicts with other networks.

When creating AzureRedisInstance, one field is mandatory: `sku.capacity`.

Optionally, you can specify the `redisConfiguration`, `redisVersion`, `shardCount`, `replicasPerPrimary` and `redisConfiguration` fields.

> > [!Note] Non SSL port is disabled.

Reviewer: ```suggestion
> [!Note] 
> Non-SSL port is disabled.
```

Author: Fixed.

| **sku.capacity**                                        | int    | Required. The service capacity of the instance. Supported values are 1, 2, 3, and 4. Note that tear is 'P' - Premium.                                                                                                                   |
| **redisVersion**                                        | int    | Optional. The version of Redis software. Defaults to `6.0`.                                                                                                                                                                             |
| **shardCount**                                          | int    | Optional. Number of shards.                                                                                                                                                                                                             |

> | **replicasPerPrimary**                                  | int    | Optional. Number of replicas, for `Premium` tier it is [up to 3.](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-high-availability)                                                                                |

Reviewer: ```suggestion
| **replicasPerPrimary**                                  | int    | Optional. Number of replicas, for the Premium tier it is [up to 3.](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-high-availability)                                                                                |
```

Author: **replicasPerPrimary**  completely removed from docs, as this option will be not be available in following revision.

> | **redisConfiguration**                                  | object | Optional.                                                                                                                                                                                                                               |

Reviewer: Can you also add a description for **redisConfiguration**? Here, it only says "Optional".

| **redisConfiguration.maxmemory-policy**                 | int    | Optional. The setting for how Redis will select what to remove when maxmemory (the size of the cache offering you selected when you created the cache) is reached. Defaults to `volatile-lru`.                                          | 
| **redisConfiguration.maxfragmentationmemory-reserved**  | int    | Optional. [Configure your maxmemory-reserved setting to improve system responsiveness.](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-best-practices-memory-management#configure-your-maxmemory-reserved-setting) |

# Auth Secret Details

The following table list the meaningful parameters of the auth Secret:

| Parameter                 | Type   | Description                                                                                            |
|---------------------------|--------|--------------------------------------------------------------------------------------------------------|

> | **.metadata.name**        | string | Name of the auth Secret. It will share the name with the AzureRedisInstance unless specified otherwise |

Reviewer: ```suggestion
| **.metadata.name**        | string | Name of the auth Secret. It shares the name with AzureRedisInstance unless specified otherwise |
```

Author: Fixed.

# AzureRedisInstance Custom Resource
The `AzureRedisInstance.cloud-resources.kyma-project.io` is a namespace-scoped custom resource (CR).
It describes the Azure Cache for Redis instance.
Once the instance is provisioned, a Kubernetes Secret with endpoint and credential details is provided in the same namespace.
By default, the created auth Secret has the same name as AzureRedisInstance.

The current implementation supports the Premium tier, which is explained in detail on the [Azure Cache for Redis overview page](https://azure.microsoft.com/en-us/products/cache).

> [!TIP] _Only for advanced cases of network topology_
> 

> > Redis requires 2 IP addresses per shard. The Premium tear supports [up to 3 shards.](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-high-availability)

Reviewer: ```suggestion
> Redis requires 2 IP addresses per shard. The Premium tier supports [up to 3 shards.](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-high-availability)
```

|--------------------------------------------------------|--------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **ipRange**                                            | object | Optional. IpRange reference. If omitted, the default IpRange is used. If the default IpRange does not exist, it will be created.                                                                                                        |
| **ipRange.name**                                       | string | Required. Name of the existing IpRange to use.                                                                                                                                                                                          | 

> | **sku.capacity**                                       | int    | Required. The service capacity of the instance. Supported values are 1, 2, 3, and 4. Note that tear is 'P' - Premium.                                                                                                                   |

Reviewer: ```suggestion
| **sku.capacity**                                       | int    | Required. The service capacity of the instance. Supported values are 1, 2, 3, and 4. Note that tier is 'P' - Premium.                                                                                                                   |
```

| **redisConfiguration.maxclients**                      | int    | Optional. Max number of Redis clients. Limited to [7,500 to 40,000.](https://azure.microsoft.com/en-us/pricing/details/cache/)                                                                                                          |
| **redisConfiguration.maxmemory-reserved**              | int    | Optional. [Configure your maxmemory-reserved setting to improve system responsiveness.](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-best-practices-memory-management#configure-your-maxmemory-reserved-setting) |
| **redisConfiguration.maxmemory-delta**                 | int    | Optional. Gets or sets value in megabytes reserved for non-cache usage per shard e.g. failover.                                                                                                                                         | 

> | **redisConfiguration.maxmemory-policy**                | int    | Optional. The setting for how Redis will select what to remove when maxmemory (the size of the cache offering you selected when you created the cache) is reached. Defaults to `volatile-lru`.                                          | 

Reviewer: ```suggestion
| **redisConfiguration.maxmemory-policy**                | int    | Optional. The setting for how Redis will select what to remove when **maxmemory** (the size of the cache offering you selected when you created the cache) is reached. Defaults to `volatile-lru`.                                          | 
```
